### PR TITLE
Toolbar

### PR DESCRIPTION
**Pull request by muganwas**
#218

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -13,6 +13,32 @@ html, body, #root, #app {
   font-size: 16px;
 }
 
+.toolbar {
+  position: absolute;
```
Comment (raineorshine) [inline src/App.css]: The toolbar should be `position: fixed`.

File: src/components/Icon.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+const Icon = props => {
+    const {
+        onclick,
+        id,
+    } = props;
```
Comment (raineorshine) [inline src/components/Icon.js]: I suggest using object deconstruction directly in the function arguments:

```js
const Icon = ({ onclick, id }) => {
```

File: src/components/Toolbar.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+const toolBar = props => {
+    return <div className="toolbar">
+        { shortcutIds.map( id=>{
+            //console.log(shortcutById(id))
+            const shortcutInfo = shortcutById(id);
+            const {
+                name,
+                exec
+            } = shortcutInfo;
```
Comment (raineorshine) [inline src/components/Toolbar.js]: These two statements can be combined.

File: src/components/Icon.js [hunk 4]
```diff
@@ -1,11 +1,7 @@
...
 
-const Icon = props => {
-    const {
-        onclick,
-        id,
-    } = props;
-    return <div className="toolbar-icon" onClick={ ()=>onclick(id)} >
+const Icon = ({ onclick, id }) => {
+    return <div className="toolbar-icon" onClick={ () => onclick(id)} >
         <svg version="1.1" className="icon" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 19.481 19.481" enableBackground="new 0 0 19.481 19.481">
```
Comment (raineorshine) [inline src/components/Icon.js]: Can you kindly store the SVG on each shortcut object in `/src/shortcuts/`? They will end up being specific to each shortcut, and that will make it easy to add new shortcuts without updating the Toolbar. Thank you!

Comment (muganwas) [reply src/components/Icon.js]: Sure thing.

File: src/components/Icon.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const Icon = ({ onclick, id, svg }) => {
+    return <div className="toolbar-icon" onClick={ () => onclick(id)} >
+        { svg }
+    </div>
+}
+
+export default Icon
```
Comment (raineorshine) [inline src/components/Icon.js]: This is too small to be its own component now. Let's inline it in `Toolbar`.

File: src/shortcuts/cursorDown.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import { store } from '../store.js'
 
 // util
+import React from 'react'
```
Comment (raineorshine) [inline src/shortcuts/cursorDown.js]: This import should go at the top of the file. "util" is reserved for imports from `util.js`.

File: src/shortcuts/cursorForward.js [hunk 7]
```diff
@@ -1,12 +1,20 @@
+import React from 'react'
```
Comment (raineorshine) [inline src/shortcuts/cursorForward.js]: Please add a newline before line comments. 

Make sure this convention is followed in all other files as well.

File: src/shortcuts/cursorNextThought.js [hunk 8]
```diff
@@ -1,15 +1,22 @@
+import React from 'react'
 import { store } from '../store.js'
-
```
Comment (raineorshine) [inline src/shortcuts/cursorNextThought.js]: Restore newline

File: src/shortcuts/cursorDown.js (hunk 6 above)
Comment (muganwas) [reply src/shortcuts/cursorDown.js]: Sorry about that,  I must have done that hurriedly. I'll be more careful next time.

File: src/shortcuts/cursorForward.js (hunk 7 above)
Comment (muganwas) [reply src/shortcuts/cursorForward.js]: Got it

File: src/shortcuts/cursorNextThought.js (hunk 8 above)
Comment (muganwas) [reply src/shortcuts/cursorNextThought.js]: Got it

File: src/components/Icon.js (hunk 5 above)
Comment (muganwas) [reply src/components/Icon.js]: Got it

**Comment by raineorshine** [comment]
Try to name commits descriptively based on what is being changed. "code review fixes" is too vague and "29/12/19" is already captured in the timestamp. 

Take a look at the commit history of this project to get a sense of this.

**Comment by raineorshine** [comment]
Also please note that tabs are 2 spaces on this project. Make sure your editor is set up to handle this correctly. Thanks!

**Pull request merged**